### PR TITLE
Add support for git worktrees

### LINKED ISSUE #4: Does not work in a `git worktree`
To reproduce:

1. Use `git worktree` to set up a branch outside of the main Git working directory
2. Try to use `github-linker` from that folder in Code
3. See this error: 
		image

Looks like the linker might be thinking that the project root is always the `.git` parent, but with a worktree this is not the case.

### PR DESCRIPTION
**Pull request by flotwig**
Fixes #4.

In a git worktree, the `.git` of the working directory is a file that contains `gitdir: /path/to/real/git/root/.git/worktrees`. 

This PR detects if `.git` is a file and if so resolves the path within that file to the real `.git` dir.

And thanks for this extension, it's extremely useful!!!

### CODE REVIEW

**Comment by flotwig** [comment]
I need to work on this some more, it doesn't seem to use the correct ref, even though the dir is right.

**Comment by flotwig** [comment]
@gimenete Good to go now!

File: src/extension.ts [hunk 1]
```diff
@@ -43,7 +37,35 @@ function calculateURL() {
...
+
+    const baseDir = path.join(gitDir, '..')
+
+    if (fs.statSync(gitDir).isFile()) {
+        // not a normal .git dir, could be a `git worktree`, read the file to find the real root
+        const text = fs.readFileSync(gitDir).toString()
+
+        console.log('gitDir is a file, checking to see if worktree', { text })
+
+        if (text.slice(0, 8) === 'gitdir: ') {
```
Comment (gimenete) [inline src/extension.ts]:
```suggestion
        const worktreePrefix = 'gitdir: ';
        if (text.startsWith(worktreePrefix)) {
```

File: src/extension.ts [hunk 2]
```diff
@@ -43,7 +37,35 @@ function calculateURL() {
...
+            // gitdir points to worktree subdir of the real gitdir
+            gitDir = path.join(text.slice(8).trim(), '../..')
```
Comment (gimenete) [inline src/extension.ts]:
```suggestion
            gitDir = path.join(text.slice(worktreePrefix.length).trim(), '..', '..');
```

**Review by gimenete — COMMENTED** [review]
Thanks a lot for your pull request!

I've just made some suggestions. Please take a look. Basically:

- Create a variable for the prefix calculation and use `startsWith`
- Small change in `path.join` to avoid using `/` that may be incompatible in Windows.

Thanks again! 🙇

File: src/extension.ts (hunk 2 above)
Comment (flotwig) [reply src/extension.ts]: nice catch

**Comment by flotwig** [comment]
Realized last-minute that this still wasn't right - it was actually using the SHA from the branch that the main repo is currently on, not the current worktree SHA.

Added a new commit to make the worktreePath (if it exists) only get used for reading the current HEAD, then still use the real gitDir for reading the refs.

**Review by gimenete — APPROVED** [review]
@flotwig awesome, thanks!!

**Pull request merged**